### ISSUE TITLE
Long delay before OnVideoSendStatusChange is invoked for 1:1-calls

### ISSUE THREAD

**Issue by rvleeuwen256** [microsoftgraph/microsoft-graph-comms-samples#663]
**Description** 
When setting up a 1:1-call it takes very long before the OnVideoSendStatusChanged event is raised to notify that the video sending has become active.

**Expected**
The OnVideoSendStatusChanged event to be raised within 1 second just like it does when setting up a call to a meeting or setting up a group-call.

**Actual**
The OnVideoSendStatusChanged event is raised exactly 5 seconds after the call has been answered.

**Additional information**
This 5 seconds delay is present when the Bot service runs in the Azure cloud and is very consistent. When the same service is run locally and a local Teams client is called, the OnVideoSendStatusChanged event is raised almost instantly after the call has been answered.
We know that the video is routed differently for meetings/group-calls and 1:1-calls. The former are always routed via Microsoft Turn servers and the 1:1-calls are attempted to be sent directly to the endpoint using the ICE-protocol. Tracing the communication showed us that the video is sent directly from the Bot service to the endpoint for the 1:1-calls, so the ICE-protocol did successfully find a direct route without a Turn server). Because the delay is so consistent it looks like a time-out on the ICE-protocol, while a direct route is already available.
This issue does not only occur when calling a Teams client in our environment/tenant, but also at all our (>500) customers.
Another interesting observation is that the two-way audio path is open almost instantly after the call answer.

**Code snippet**
The video socket settings are created using the following code:
```
new VideoSocketSettings
                {
                    StreamDirections = StreamDirection.Sendrecv,
                    ReceiveColorFormat = VideoColorFormat.H264,
                    MediaType = MpMediaType.Video,
                    SupportedSendVideoFormats = new List<VideoFormat>()
                    {
                        VideoFormat.H264_1920x1080_30Fps,
                        VideoFormat.H264_1920x1080_15Fps,
                        VideoFormat.H264_1280x720_30Fps,
                        VideoFormat.H264_1280x720_15Fps,
                        VideoFormat.H264_960x540_30Fps,
                        VideoFormat.H264_640x360_30Fps,
                        VideoFormat.H264_640x360_15Fps,
                        VideoFormat.H264_424x240_15Fps,
                        VideoFormat.H264_320x180_15Fps,
                    },
                    MaxConcurrentSendStreams = 1,
                    CroppingMode = CroppingMode.FitToFrame,
                }
```

**NuGet Versions**
 - Microsoft.Skype.Bots.Media 1.25.0.729-alpha
 - Microsoft.Graph.Communications.Calls.Media 1.2.0.5304
 - Microsoft.Graph 4.42.0

**Comment by InDieTasten** [comment]
Can't say too much about the video socket and timing relating to that. What here's what I do know:

Running local media bots in a location separate to the calls hosting region adds substantial delay to answering calls. Upwards of 5-10 seconds.

Are you running your bot in a location close to the calls media hosting region? You can check the location as part of the ICall resource properties.

**Comment by ssulzer** [comment]
@rvleeuwen256 Can you please share a call-id? Thanks.

**Comment by rvleeuwen256** [comment]
> @rvleeuwen256 Can you please share a call-id? Thanks.

Hello ssulzer,

The following is a call made with the Bot framework to a native Windows desktop client.

Teams call Id: 5a1f5c00-4496-4c7f-a31a-c2851f92ca3f
CallChainId: 26b8c090-100a-4da1-bcde-fe9daaf20b24

Just for convenience I've added a description of the events that happened for this call according to our logging:

Answer of call creation: 2023-08-01 14:12:29.364
Comm event 'Establishing': 2023-08-01 14:12:29.411
OnAudioSendStatusChanged Active: 2023-08-01 14:12:40.717
Comm event 'Established': 2023-08-01 14:12:40.960
OnVideoSendStatusChanged Inactive: 2023-08-01 14:12:46.035
OnVideoSendStatusChanged Active: 2023-08-01 14:12:46.036

The 'Establishing' comm event is received very quickly (50 ms) after the call has been created. The AudioSendStatus is set to Active even before the 'Established' event has been received. Then it takes 5,075 seconds before the VideoSendStatus is set to Inactive and 1 ms later another event is raised to set the status to Active.

**Comment by rvleeuwen256** [comment]
We investigated the issue further and found out that it was caused by the firewall that runs on the VM. 

Using the verbose logging of the media platform and WireShark traces we found out that (contrary to what we thought) there was no direct media connection between the bot and the Teams client. The Stun binding requests that are sent from the Teams client to the bot during the Ice path discovery were blocked by the firewall. As a result the media path over a Microsoft Turn server was chosen, which immediately routes the audio, but waits a 5 seconds time-out before starting to route the video (obviously expecting a media direct connection will be found soon).

By adding an ‘allow’ firewall rule for the Udp protocol for the ‘media port range’ provided during the media platform initialization a direct media connection could be established which resulted in video in the Teams client in about 1 second.
